k9mail/src/main/java/com/fsck/k9/mailstore/migrations/MigrationTo51.java

```diff
@@ -587,20 +590,32 @@ private static MimeStructureState insertTextualPartIntoDatabase(SQLiteDatabase d
                 isHtml ? "text/html; charset=\"utf-8\"" : "text/plain; charset=\"utf-8\"");
         mimeHeader.setHeader(MimeHeader.HEADER_CONTENT_TRANSFER_ENCODING, MimeUtil.ENC_QUOTED_PRINTABLE);
 
-        ByteArrayOutputStream contentOutputStream = new ByteArrayOutputStream();
-        QuotedPrintableOutputStream quotedPrintableOutputStream =
-                new QuotedPrintableOutputStream(contentOutputStream, false);
-        quotedPrintableOutputStream.write(content.getBytes());
-        quotedPrintableOutputStream.flush();
-        byte[] contentBytes = contentOutputStream.toByteArray();
+        byte[] contentBytes;
+        int decodedBodySize;
+        int dataLocation;
+        if (content != null) {
+            ByteArrayOutputStream contentOutputStream = new ByteArrayOutputStream();
+            QuotedPrintableOutputStream quotedPrintableOutputStream =
+                    new QuotedPrintableOutputStream(contentOutputStream, false);
+            quotedPrintableOutputStream.write(content.getBytes());
+            quotedPrintableOutputStream.flush();
```

Comment:
General note: Closing a stream will also flush the output. It has the additional benefit of being the right thing to do and not being flagged by static analyzers or runtime checks.